runtime/src/accounts.rs

```diff
@@ -1047,30 +1047,17 @@ impl Accounts {
             let (execution_result, nonce) = &execution_results[i];
             let maybe_nonce = match (execution_result, nonce) {
                 (Ok(_), Some(nonce)) => {
-                    let address = nonce.address();
-                    let account = nonce.account();
-                    let maybe_fee_payer_account = nonce.fee_payer_account();
-                    Some((address, account, maybe_fee_payer_account, true))
-                }
-                (Err(TransactionError::InstructionError(index, _)), Some(nonce)) => {
-                    let nonce_marker_ix_failed = if merge_nonce_error_into_system_error {
-                        // Don't advance stored blockhash when the nonce marker ix fails
-                        *index == NONCED_TX_MARKER_IX_INDEX
+                    if leave_nonce_on_success {
+                        None
                     } else {
-                        false
-                    };
-                    let address = nonce.address();
-                    let account = nonce.account();
-                    let maybe_fee_payer_account = nonce.fee_payer_account();
-                    Some((
-                        address,
-                        account,
-                        maybe_fee_payer_account,
-                        !nonce_marker_ix_failed,
-                    ))
+                        Some((nonce, false /* rollback */))
```

Comment:
```suggestion
                        Some((nonce, false /* don't rollback stored nonce */))
```

It's not clear on first glance that this comment is simply referring to the variable name, so let's just be verbose